src/features/room/sagas.ts

```diff
@@ -0,0 +1,95 @@
+import axios, { AxiosResponse } from 'axios'
+import { Task } from 'redux-saga'
+import {
+  all,
+  call,
+  cancel,
+  delay,
+  fork,
+  put,
+  take,
+  takeLatest,
+} from 'redux-saga/effects'
+
+import {
+  RoomModel,
+  getMatchedRoom,
+  keepRoomAlive,
+} from '../../services/roomService'
+import config from '../../utils/config'
+import { ServiceResponse } from '../../utils/types'
+import { Question } from '../questionBank/types'
+import { closeRoom, openRoom, setQuestionData } from './slice'
+import { RoomSagaActions } from './types'
+
+const keepAliveIntervalInMs = 5000
+const retryFindMatchRoomDelayInMs = 3000
+
+function* startLoadMatchRoom() {
```

Comment:
Suggestion:
maybe this could be handled better by using `yield race` instead of forking another function, can refer to the `feature-services-integration` branch for the matching sagas.ts file for reference
can also refer to this link for saga docs: https://redux-saga.js.org/docs/advanced/RacingEffects

From the code, I'm guessing the room is trying to call the function to load the matched room, and if its not found it will be retrying until theres a matched room, then the component itself will dispatch a stop load room? essentially with yield race you can call loadMatchRoomTask and add a another effect called "cancel" that will stop loadMatchRoomTask

can try it but if it doesnt work for some reason can just leave it be ba 😬